client/blocks/eligibility-warnings/hold-list.tsx

```diff
@@ -155,7 +156,14 @@ export const HardBlockingNotice = ( {
 	const blockingHold = holds.find( ( h ): h is keyof ReturnType< typeof getBlockingMessages > =>
 		isHardBlockingHoldType( h, blockingMessages )
 	);
-	if ( ! blockingHold ) {
+
+	/*
+		For Atomic sites on plans below Business it will return the holds TRANSFER_ALREADY_EXISTS and NO_BUSINESS_PLAN.
+		Because TRANSFER_ALREADY_EXISTS is present and 'blocking' it will show an "Upload in progress" notice even when there isn't one.
+		In this scenario we need to check if it's an Atomic ste (TRANSFER_ALREADY_EXISTS) on a plan below Business (NO_BUSINESS_PLAN)
+		so we can stop the render of "Upload in progress" and prompt them to upgrade instead.
+	*/
+	if ( ! blockingHold || isAtomicSiteWithoutBusinessPlan( holds ) ) {
```

Comment:
Minor and non-blocking: If this check is done within `isHardBlockingHoldType` you can avoid repeating the same check below. You'd need to change the signature of the `isHardBlockingHoldType` function so the list of holds is passed as an extra param, but I think this improves the readability.

Reply:
The `isHardBlockingHoldType` function is also used in `hasBlockingHold` which is also imported elsewhere. I will see if I can understand this functions usages elsewhere with enough confidence to move this check but this was my original reason for keeping it separate.